pkg/backend/backend_test.go

```diff
@@ -0,0 +1,457 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/pulumi/pulumi/pkg/engine"
+	"github.com/pulumi/pulumi/pkg/operations"
+	"github.com/pulumi/pulumi/pkg/resource"
+	"github.com/pulumi/pulumi/pkg/resource/config"
+	"github.com/pulumi/pulumi/pkg/resource/deploy"
+	"github.com/pulumi/pulumi/pkg/tokens"
+	"github.com/pulumi/pulumi/pkg/util/result"
+)
+
+func TestGetStackResourceOutputs(t *testing.T) {
+	// Create a `backendClient` that consults a (mock) `Backend` to make sure it can get the stack
+	// resource outputs correctly.
+
+	resc1 := liveState("liveType1", "resc1", resource.PropertyMap{
```

Comment:
NIT: I think it would be ideal if the URN And type followed the structure implied by the URN and Type tokens (i.e. they looked more like the real URNs and Types we'd see generated from real programs).

Reply:
Done.